### ISSUE TITLE
stylus not working in fullscreen mode

### ISSUE THREAD

**Issue by Scaby79** [xournalpp/xournalpp#5852]
### Operating System

Linux

### (Linux only) Distribution

Ubuntu 24.04.1

### (Linux only) Desktop Environment

Gnome

### (Linux Only) Display Server

Wayland

### Installation Method

Official Release (stable)

### Xournal++ Version

1.2.3

### libgtk Version

3.24.41

### Bug Description

Stylus input stops working when entering fullscreen or presentation mode.
If the option drawing with touchscreen is enabled handwriting works in fullscreen mode but stylus still is not working.
It is working again after leaving fullscreen or presentation mode.

### Expected Behaviour

it is possible to use xournal in fullscreen with stylus

### Steps to Reproduce

1. open journal not in fullscreen and not presentation mode
2. write with stylus
3. enter fullscreen or presentation mode
4. try to write with stylus

**Comment by Scaby79** [comment]
Yes, most likely the combination of xournalpp and wayland is causing the problem. 
In xorg it is working für me to BUT using xorg I'm running into another bug -> #5858 

I would prefer xorg over wayland but I really do need the handwriting option as I use xournal with an external smartboard as input.

**Comment by ckoever** [comment]
Same problem for me. The pen seems to work a short while in full screen, but then suddenly stops. For me, the touchscreen also stops working in full screen mode.

**Comment by stendarr** [comment]
Can confirm the same issue on Debian testing with xournalpp 1.2.3 on wayland.

@ckoever When I enable fullscreen (with F11) while writing, I can keep writing while the stylus remains close to the screen (with the pointer still hovering around) but as soon as I remove the stylus further, it stops working. Is it the same for you?

**Comment by Scaby79** [comment]
@stendarr I can confirm the behaviour you described: The styles still works as long as it is close to the surface after entering fullscreen mode. It stops working as soon as it is not near enough to the surface and bringing the stylus back to the surface does not get it back to work again.

**Comment by stendarr** [comment]
I'm not able to test with other DEs at the moment but I can confirm the same issue happens regardless of the program running in fullscreen, so it's definitely not a xournalpp (only) issue.

**Comment by stendarr** [comment]
Looks like this is an issue with [mutter](https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3956#note_2227516), it has been [resolved in version 46.5](https://gitlab.gnome.org/GNOME/mutter/-/blob/gnome-46/NEWS#L9). Now we just have to wait for our respective distros to package it, I guess.

**Comment by swsnr** [comment]
I no longer have this issue with GNOME 47, which according to its NEWS file includes the same fix as GNOME 46.5

**Comment by Dr0gelix** [comment]
So thankfully I am not alone with this problem, and besides the LTS version that there will be no mutter, and/or GNOME update until next LTS or going to the .10 version. My brother came across another solution or a problem what causes it.

We both use the gnome extension Hide Top Bar by mathieu.bidon.ca. If it is set to off, xournalpp works fine in fullscreen mode. On the other hand, in presentation mode via F11 it doesn't work again. So, at least for us, a current solution is not to use this extension currently or set it off occasionally.

**Comment by crazy-to-bike** [comment]
I have the same problem. Unfortunately Ubuntu 24.04 is still using Mutter / Gnome 46.2 and I can't find a roadmap, when or if there will be 46.5 at Ubuntu 24.04

**Comment by crefoo** [comment]
Ubuntu 24.04.1 LTS Still having the same issue, I might just switch back to windows at this point.

**Comment by Scaby79** [comment]
In Ubuntu 24.10. this is no issue anymore.

**Comment by DavidBraun1** [comment]
Does this mean, that it is going to be fixed in the next LTS?

**Comment by SnailSale** [comment]
Anyone else having this issue with again using Wayland Mutter,  Gnome 50?

When Xournal++ is in full screen mode: 
- IF a proximity-out event of the stylus occurs **on the Xournal++ window** then the stylus prox-in event is **only detected outside the Xournal++ window (taskbar, panel, dash)**
- IF the prox-out event happened **outside the Xournal++ window** (taskbar, panel, dash) then the stylus prox-in event is always detected (also on the Xournal++ window).

Tested with  Xournal++ versions 1.3.4-1.1 & 1.3.2-2.2

**Comment by fabioschoen** [comment]
Just upgraded to Ubuntu 26.04 on a Lenovo Yoga 9 2-in-1 14IMH9
With a Bamboo Ink Plus pen still the same bug appears, same behaviour on Xournal++ 1.3.3  (and also on OpenBoard)

**Comment by SnailSale** [comment]
@fabioschoen: 
I did some deeper digging, and it seems to be an issue caused by gnome/wayland (with Krita the same error can be observed).
For me a ghost-cursor can be seen in some cases always stuck to the upper left corner. If I rescale the xournal++ (or other app) so that it does not touch the left side of the desctop the program works flawlessly.

-> Not a xournal++  issue

**Comment by fabioschoen** [comment]
Many  thanks for your kind reply! Indeed, with xournal++ switching from
full screen to window works, although it is a bit uncomfortable. Other
applications do not allow you to do so. I am using openboard, and there
seem to be no way to get the same result....
However, I noticed that if I switch back and forth between single and multi
screen mode, eventually the pen works.. And if it get stuck, an Alt-Tab to
a different window and back will keep it working... Not ideal, but I might
be able to live with this...
fabio

**Comment by GithubAnon0000** [comment]
> Not ideal, but I might
> be able to live with this...

Please also head over to the gnome team and report the bug. They can only fix it if they know about the bug. And maybe that means you won't have to live with this bug for a long time.

**Comment by abc8747** [comment]
I have the same issue when using Xournal++ and Krita on Ubuntu 26.04.1 LTS (mutter `50.1-0ubuntu2.2`).

After some digging it appears to be a mutter regression: https://gitlab.gnome.org/GNOME/mutter/-/work_items/4803 affecting mutter 50.0..50.2. It should be fixed in mutter 50.3.

**Comment by fabioschoen** [comment]
Indeed! Meanwhile I am running the application in a window, instead of full
screen...
fabio

**Comment by Liftyee** [comment]
With the help of Gemini I made a one line patch to my Ubuntu 24.04 mutter to fix this. I can provide it if it is useful.
